### ISSUE TITLE
Divide by zero errors?

### ISSUE THREAD

**Issue by FalafelGood** [clementsw/interferometer#2]
Hi there, 

I'm trying to perform a square decomposition of a unitary matrix that corresponds to a type-1 fusion operation (happy to give more context if necessary), but I seem to be dividing by zero. Here's some code for reproducing the behavior.

```
import numpy as np
import interferometer as itf

type1fusion = 1/np.sqrt(2) * np.matrix([[1,0,0,1],[0,np.sqrt(2),0,0],[1,0,0,-1],[0,0,np.sqrt(2),0]])
I = itf.square_decomposition(type1fusion)
print(I.BS_list)
```

When I run this code I first get a cascade of warnings:

"RuntimeWarning: invalid value encountered in double_scalars
  theta = np.arctan(abs(U[N-1-jj, ii-jj] / U[N-1-jj, ii-jj+1]))"

The theta and phi values of the beamsplitters are all NaN.

Is this an issue with the code, or am I implicitly trying to do something that's actually impossible?

Cheers

**Comment by FalafelGood** [comment]
Update: I've identified that this issue occurs because there are zeros in the unitary matrix I want to decompose. It's a bit hacky, but I've found that substituting these with small positives close to machine precision seems to produce good beamsplitter decompositions.

@clementsw, Is this something I can fix for you, and if so how would you like this done?

**Comment by clementsw** [comment]
Thanks for pointing this out @FalafelGood ! I've reproduced the error, and have suggested a fix in #3 . Would you mind trying that out and confirming that this fixes things for you?

**Comment by FalafelGood** [comment]
Hey, thanks for that @clementsw it's working great!
